src/core/optimizer.ts

```diff
@@ -218,7 +228,7 @@ export function optimizer(
 
     // Per the WebAssembly JS API spec, the result is only an array if there's
     // more than one return value. There's no way to make it always an array.
```

Comment:
could we just always add a 0 as the first return value?

Reply:
It should be moot once we move the parameters into memory. Then we can read them directly from JS, and optimize won't return any value.